### PR TITLE
feat: Update pipelines controller

### PR DESCRIPTION
**Pull request by maciaszczykm**
Zrzut ekranu 2024-01-8 o 16 41 29

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206268847103194

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Registered pipeline reconciler, so marking is as draft for now.

File: controller/config/samples/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      gates:
+        - name: job-gate
+          type: JOB
+          clusterRef:
+            name: dev
+            namespace: default
+          spec:
+            job:
+              namespace: default
+# TODO: Uncomment after fixing struct issue.
```
Comment (maciaszczykm) [inline controller/config/samples/pipeline.yaml:135]: This one needs to be solved before merging.

**Review by zreigz — APPROVED** [review]

**Review by michaeljguarino — APPROVED** [review]
lgtm

**Comment by maciaszczykm** [comment]
Unit tests and lint are failing on master. We'll have to fix them from there.

**Pull request merged**